File: tests/Generator/resource/ruby/complex/struct_definition_type.rb

# A struct represents a class/structure with a fix set of defined properties
class StructDefinitionType
  extend DefinitionType
  attr_accessor :base, :discriminator, :mapping, :parent, :properties, :type

  def initialize(base, discriminator, mapping, parent, properties, type)
    @base = base
    @discriminator = discriminator
    @mapping = mapping
    @parent = parent
    @properties = properties
    @type = type
  end
end

File: tests/Generator/resource/kotlin/complex/StructDefinitionType.kt

import com.fasterxml.jackson.annotation.*

/**
 * A struct represents a class/structure with a fix set of defined properties.
 */
open class StructDefinitionType : DefinitionType {
    @JsonProperty("base") var base: Boolean? = null
    @JsonProperty("discriminator") var discriminator: String? = null
    @JsonProperty("mapping") var mapping: HashMap<String, String>? = null
    @JsonProperty("parent") var parent: ReferencePropertyType? = null
    @JsonProperty("properties") var properties: HashMap<String, PropertyType>? = null
}

